test/test_buffer_protocol.py

```diff
@@ -0,0 +1,175 @@
+            for j in range(SIZE - i + 1):
+                self._run_test(SHAPE, dtype, count=i, first=j)
+
+    @onlyCPU
+    @dtypes(*common.torch_to_numpy_dtype_dict.keys())
+    def test_invalid_positional_args(self, device, dtype):
+        bytes = get_dtype_size(dtype)
+        in_bytes = SIZE * bytes
+        # Empty array
+        with self.assertRaisesRegex(ValueError,
+                                    r"both buffer length \(0\) and count"):
+            empty = numpy.array([])
+            torch.frombuffer(empty)
+        # Count equals 0
+        with self.assertRaisesRegex(ValueError,
+                                    r"both buffer length .* and count \(0\)"):
+            self._run_test(SHAPE, dtype, count=0)
+        # Offset negative and bigger than total length
+        with self.assertRaisesRegex(ValueError,
+                                    rf"offset \(-{bytes} bytes\) must be"):
+            self._run_test(SHAPE, dtype, first=-1)
+        with self.assertRaisesRegex(ValueError,
+                                    rf"offset \({in_bytes} bytes\) must be .* "
+                                    rf"buffer length \({in_bytes} bytes\)"):
+            self._run_test(SHAPE, dtype, first=SIZE)
+        # Non-multiple offset with all elements
+        if bytes > 1:
+            offset = bytes - 1
+            with self.assertRaisesRegex(ValueError,
+                                        rf"buffer length \({in_bytes - offset} bytes\) after "
+                                        rf"offset \({offset} bytes\) must be"):
+                self._run_test(SHAPE, dtype, offset=bytes - 1)
+        # Count too big for each good first element
+        for first in range(SIZE):
+            count = SIZE - first + 1
+            with self.assertRaisesRegex(ValueError,
+                                        rf"requested buffer length \({count} \* {bytes} bytes\) "
+                                        rf"after offset \({first * bytes} bytes\) must .*"
+                                        rf"buffer length \({in_bytes} bytes\)"):
+                self._run_test(SHAPE, dtype, count=count, first=first)
+
+    @onlyCPU
+    @dtypes(*common.torch_to_numpy_dtype_dict.keys())
+    def test_shared_buffer(self, device, dtype):
+        x = common.make_tensor((1,), device, dtype)
+        # Modify the whole tensor
+        arr, tensor = self._run_test(SHAPE, dtype)
+        tensor[:] = x
+        self.assertEqual(arr, tensor)
+        self.assertTrue((tensor == x).all().item())
+
+        # Modify the whole tensor from all valid offsets, given
+        # a count value
+        for count in range(-1, SIZE + 1):
+            if count == 0:
+                continue
+
+            actual_count = count if count > 0 else SIZE
+            for first in range(SIZE - actual_count):
+                last = first + actual_count
+                arr, tensor = self._run_test(SHAPE, dtype, first=first, count=count)
+                tensor[:] = x
```

Comment:
What about editing a value in the array and ensuring that appears properly in the tensor?

Reply:
That makes sense. I will add it.